resctrl/utils.go

```diff
@@ -0,0 +1,285 @@
+	if stat, _ := os.Stat(monGroupPath); stat == nil {
+		err = os.Mkdir(monGroupPath, os.ModePerm)
+	}
+	if err != nil {
+		return "", err
+	}
+
+	for _, pid := range pids {
+		threadFiles, err := ioutil.ReadDir(filepath.Join(processPath, pid, processTask))
+		if err != nil {
+			return "", err
+		}
+
+		processThreads, err := getAllProcessThreads(threadFiles)
+		if err != nil {
+			return "", err
+		}
+		for _, thread := range processThreads {
+			err = intelrdt.WriteIntelRdtTasks(monGroupPath, thread)
+			if err != nil {
+				secondError := os.Remove(monGroupPath)
+				if secondError != nil {
+					return "", fmt.Errorf("%v : %v", err, secondError)
+				}
+				return "", err
+			}
+		}
+	}
+
+	return monGroupPath, nil
+}
+
+func getPids(containerName string) ([]int, error) {
+	if containerName == "" {
+		return nil, fmt.Errorf(noContainerError)
+	}
+	pids, err := cgroups.GetAllPids(filepath.Join(pidsPath, containerName))
+	if err != nil {
+		return nil, fmt.Errorf("couldn't obtain pids: %v", err)
+	}
+	return pids, nil
+}
+
+func getAllProcessThreads(threadFiles []os.FileInfo) ([]int, error) {
+	processThreads := make([]int, 0)
+	for _, file := range threadFiles {
+		pid, err := strconv.Atoi(file.Name())
+		if err != nil {
+			return nil, fmt.Errorf("couldn't parse %q file: %v", file.Name(), err)
+		}
+		processThreads = append(processThreads, pid)
+	}
+
+	return processThreads, nil
+}
+
+func findControlGroup(pids []string) (string, error) {
+	if len(pids) == 0 {
+		return "", fmt.Errorf(noPidsError)
+	}
+	availablePaths, err := filepath.Glob(filepath.Join(rootResctrl, "*"))
+	if err != nil {
+		return "", err
+	}
+
+	for _, path := range availablePaths {
+		switch path {
+		case filepath.Join(rootResctrl, cpusFileName):
+			continue
+		case filepath.Join(rootResctrl, cpusListFileName):
+			continue
+		case filepath.Join(rootResctrl, infoDirName):
+			continue
+		case filepath.Join(rootResctrl, monDataDirName):
+			continue
+		case filepath.Join(rootResctrl, monGroupsDirName):
+			continue
+		case filepath.Join(rootResctrl, schemataFileName):
+			continue
+		case filepath.Join(rootResctrl, tasksFileName):
+			inGroup, err := checkControlGroup(rootResctrl, pids)
+			if err != nil {
+				return "", err
+			}
+			if inGroup {
+				return rootResctrl, nil
+			}
+		default:
+			inGroup, err := checkControlGroup(path, pids)
+			if err != nil {
+				return "", err
+			}
+			if inGroup {
+				return path, nil
+			}
+		}
+	}
+
+	return "", fmt.Errorf("there is no available control group")
+}
+
+func checkControlGroup(path string, pids []string) (bool, error) {
+	if len(pids) == 0 {
+		return false, fmt.Errorf(noPidsError)
+	}
+
+	taskFile, err := fscommon.ReadFile(path, "tasks")
+	if err != nil {
+		return false, fmt.Errorf("couldn't obtain pids: %v", err)
+	}
+
+	for _, pid := range pids {
+		if !strings.Contains(taskFile, pid) {
+			return false, nil
+		}
+	}
+
+	return true, nil
+}
+
+func readStat(path string) (uint64, error) {
```

Comment:
`readStat` suggest that you want to read from a file and stat it in the same time. What `path` represents? I guess you read some statistics from some file (too many "some").

Reply:
perhaps: `readStatisticFrom(file string) (uint64, error)` would be better?